HealthcareApp/src/Healthcare.Infrastructure/Repository/ScheduleRepository.cs

```diff
@@ -0,0 +1,62 @@
+using Application.Abstractions;
+using Domain.Entities;
+using Healthcare.Application.DTOs.Schedule;
+using Healthcare.Infrastructure.Persistance;
+using Microsoft.EntityFrameworkCore;
+
+namespace Infrastructure.Repository;
+
+public class ScheduleRepository : IScheduleRepository
+{
+    private readonly AppDbContext _appDbContext;
+
+    public ScheduleRepository(AppDbContext appDbContext) => _appDbContext = appDbContext;
+
+    public async Task<List<Schedule>> GetDoctorSchedule(string doctorId, int pageSize)
+        => await _appDbContext.Schedules.Where(s => s.DoctorId == doctorId).Take(pageSize).ToListAsync();
+
+    public async Task CreateDefaultWorkingScheduleAsync(List<Schedule> workingHours)
+    {
+        _appDbContext.Schedules.AddRange(workingHours);
+        await _appDbContext.SaveChangesAsync();
+    }
+
+    public async Task<Dictionary<string, List<TimeSlotDTO>>> GetTimeSlots(ApplicationUser doctor, List<Schedule> schedules)
+    {
+        var weekSlots = new Dictionary<string, List<TimeSlotDTO>>();
+
+        foreach (var day in Enum.GetValues<DayOfWeek>())
+        {
+            var schedule = schedules.FirstOrDefault(s => s.DayOfWeek == day);
+            if (schedule == null) continue;
+
+            var allAppointments = await _appDbContext.Appointments
+                .Where(a => a.DoctorId == doctor.Id)
+                .ToListAsync();
+            
+            var appointments = allAppointments
+                .Where(a => a.AppointmentDate.DayOfWeek == day)
+                .Select(a => new { a.StartTime, a.EndTime })
+                .ToList();
```

Comment:
What is the problem doing the filtration on DB level? Why do you load all appointments of a doctor in memory?

Reply:
`var appointments =   _appDbContext.Appointments
                .Where(a => a.DoctorId == doctor.Id)
                .ToList()
                .Where(a => a.AppointmentDate.DayOfWeek == day)
                .Select(a => new { a.StartTime, a.EndTime })
                .ToList();` is it ok for replacement? others it throws invalid operation exception

Reply:
No it's not ok, because you do
`_appDbContext.Appointments.Where(a => a.DoctorId == doctor.Id).ToList()` and this will load all doctor appointments.